marker_adapter/src/loader.rs

```diff
@@ -25,9 +27,17 @@ impl LintCrateInfo {
     pub fn list_from_env() -> Result<Vec<LintCrateInfo>, AdapterError> {
         let env_str = std::env::var_os(LINT_CRATES_ENV).ok_or(AdapterError::LintCratesEnvUnset)?;
 
-        Ok(std::env::split_paths(&env_str)
-            .map(|path| LintCrateInfo { path })
-            .collect())
+        let mut lint_crates = vec![];
+        for item in env_str.to_str().ok_or(AdapterError::LintCratesEnvMalformed)?.split(';') {
+            let mut item_parts = item.splitn(2, ':');
+            let name = item_parts.next().ok_or(AdapterError::LintCratesEnvMalformed)?;
+            let path = item_parts.next().ok_or(AdapterError::LintCratesEnvMalformed)?;
```

Comment:
This feels like there should be a cleaner way to handle these errors.

Reply:
I was thinking about going full serde. I think having a more rigorous definition of this environment variable could be useful, especially if this is meant to extend to other drivers in the eventual future.

It just felt like more than what we were trying to accomplish here, so I held back on serde-ifying.

Reply:
If you want to, go full serde on it :D

Reply:
I'm not willing to commit to that right now. but I think I might poke at it a bit as a separate experiment later.